diff --git a/key/aziot-key-openssl-engine-shared/Cargo.toml b/key/aziot-key-openssl-engine-shared/Cargo.toml
index 583c218..eb50292 100644
--- a/key/aziot-key-openssl-engine-shared/Cargo.toml
+++ b/key/aziot-key-openssl-engine-shared/Cargo.toml
@@ -12,7 +12,7 @@ crate-type = ["cdylib"]
 
 
 [dependencies]
-openssl-errors = "0.1"
+openssl-errors = "0.2"
 openssl-sys = "0.9"
 url = "2"
 
diff --git a/key/aziot-key-openssl-engine/Cargo.toml b/key/aziot-key-openssl-engine/Cargo.toml
index 1d0b5c2..0950c38 100644
--- a/key/aziot-key-openssl-engine/Cargo.toml
+++ b/key/aziot-key-openssl-engine/Cargo.toml
@@ -12,7 +12,7 @@ base64 = "0.13"
 foreign-types-shared = "0.1"
 log = "0.4"
 openssl = "0.10"
-openssl-errors = "0.1"
+openssl-errors = "0.2"
 openssl-sys = "0.9"
 
 aziot-key-client = { path = "../aziot-key-client" }
