diff --git a/edgelet/aziot-edged/Cargo.toml b/edgelet/aziot-edged/Cargo.toml
index 4ff0c9125..97790fd68 100644
--- a/edgelet/aziot-edged/Cargo.toml
+++ b/edgelet/aziot-edged/Cargo.toml
@@ -4,6 +4,7 @@ version = "0.1.0"
 authors = ["Azure IoT Edge Devs"]
 publish = false
 edition = "2018"
+repository = "https://github.com/Azure/iotedge"
 
 [dependencies]
 base64 = "0.9"
diff --git a/edgelet/iotedge/Cargo.toml b/edgelet/iotedge/Cargo.toml
index c1a9c6b58..6c987e00b 100644
--- a/edgelet/iotedge/Cargo.toml
+++ b/edgelet/iotedge/Cargo.toml
@@ -2,6 +2,7 @@
 name = "iotedge"
 version = "0.1.0"
 authors = ["Azure IoT Edge Devs"]
+repository = "https://github.com/Azure/azure-iotedge"
 description = """
 The iotedge tool is used to manage the IoT Edge runtime.
 """
diff --git a/edgelet/iotedge/src/check/mod.rs b/edgelet/iotedge/src/check/mod.rs
index 44b5df5c2..041fdc6d5 100644
--- a/edgelet/iotedge/src/check/mod.rs
+++ b/edgelet/iotedge/src/check/mod.rs
@@ -237,7 +237,7 @@ impl Check {
             description: String,
             result: CheckResult,
             additional_info: serde_json::Value,
-        };
+        }
 
         let mut checks: BTreeMap<String, CheckOutputSerializable> = Default::default();
 
