diff --git a/tpm/aziot-tpmd/src/lib.rs b/tpm/aziot-tpmd/src/lib.rs
index 4094463..35cbd71 100644
--- a/tpm/aziot-tpmd/src/lib.rs
+++ b/tpm/aziot-tpmd/src/lib.rs
@@ -256,7 +256,8 @@ impl Api {
                 Persistent::OWNER_HIERARCHY,
                 &auth_key_handle,
                 &Persistent::PASSWORD_SESSION,
-                0x8100_1000,
+                //very dirty fix, which prevents configuring auth_key_index in config.toml
+                tss_minimal::handle::PERSISTENT_OBJECT_BASE + 0x00_00_01,
             )?
             .expect("Esys_EvictControl with a transient handle returns a persistent handle");
 
