diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7bbfa6f3..9d1989b0 100644
--- a/tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/c-shared/CMakeLists.txt
+++ b/tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/c-shared/CMakeLists.txt
@@ -176,7 +176,7 @@ elseif(LINUX) #LINUX OR APPLE
     endif()
     # Turn off warning that can not be fixed right now
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-variable -Wno-missing-braces -Wno-missing-field-initializers -Wno-format-nonliteral")
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-variable -Wno-missing-braces -Wno-missing-field-initializers -Wno-format-nonliteral")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-variable -Wno-missing-braces -Wno-missing-field-initializers -Wno-format-nonliteral -Wno-array-parameter -Wno-deprecated-declarations")
 elseif(APPLE)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wformat-security")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wformat-security")

diff --git a/adapters/tlsio_openssl.c b/adapters/tlsio_openssl.c
index a73c129c..c621c861 100644
--- a/tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/c-shared/adapters/tlsio_openssl.c
+++ b/tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/c-shared/adapters/tlsio_openssl.c
@@ -866,7 +866,7 @@ static int add_certificate_to_store(TLS_IO_INSTANCE* tls_io_instance, const char
         }
         else
         {
-#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && (OPENSSL_VERSION_NUMBER < 0x20000000L)
+#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && (OPENSSL_VERSION_NUMBER <= 0x30000030L)
             const BIO_METHOD* bio_method;
 #else
             BIO_METHOD* bio_method;

diff --git a/adapters/x509_openssl.c b/adapters/x509_openssl.c
index 11be7a03..96993f40 100644
--- a/tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/c-shared/adapters/x509_openssl.c
+++ b/tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/c-shared/adapters/x509_openssl.c
@@ -73,7 +73,7 @@ static int load_certificate_chain(SSL_CTX* ssl_ctx, const char* certificate)
                 // certificates.
 
                 /* Codes_SRS_X509_OPENSSL_07_006: [ If successful x509_openssl_add_ecc_credentials shall to import each certificate in the cert chain. ] */
-#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && (OPENSSL_VERSION_NUMBER < 0x20000000L)
+#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && (OPENSSL_VERSION_NUMBER <= 0x30000030L)
                 SSL_CTX_clear_extra_chain_certs(ssl_ctx);
 #else
                 if (ssl_ctx->extra_certs != NULL)
@@ -264,7 +264,7 @@ int x509_openssl_add_certificates(SSL_CTX* ssl_ctx, const char* certificates)
         else
         {
             /*Codes_SRS_X509_OPENSSL_02_012: [ x509_openssl_add_certificates shall get the memory BIO method function by calling BIO_s_mem. ]*/
-#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && (OPENSSL_VERSION_NUMBER < 0x20000000L)
+#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && (OPENSSL_VERSION_NUMBER <= 0x30000030L)
             const BIO_METHOD* bio_method;
 #else
             BIO_METHOD* bio_method;
-- 
2.30.2

diff --git a/tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/utpm/deps/c-utility/CMakeLists.txt b/tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/utpm/deps/c-utility/CMakeLists.txt
index 7bbfa6f3..9d1989b0 100644
--- a/tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/utpm/deps/c-utility/CMakeLists.txt
+++ b/tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/utpm/deps/c-utility/CMakeLists.txt
@@ -176,7 +176,7 @@ elseif(LINUX) #LINUX OR APPLE
     endif()
     # Turn off warning that can not be fixed right now
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-variable -Wno-missing-braces -Wno-missing-field-initializers -Wno-format-nonliteral")
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-variable -Wno-missing-braces -Wno-missing-field-initializers -Wno-format-nonliteral")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-variable -Wno-missing-braces -Wno-missing-field-initializers -Wno-format-nonliteral -Wno-array-parameter -Wno-deprecated-declarations")
 elseif(APPLE)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wformat-security")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wformat-security")

diff --git a/tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/utpm/deps/c-utility/adapters/tlsio_openssl.c b/tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/utpm/deps/c-utility/adapters/tlsio_openssl.c
index a73c129c..c621c861 100644
--- a/tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/utpm/deps/c-utility/adapters/tlsio_openssl.c
+++ b/tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/utpm/deps/c-utility/adapters/tlsio_openssl.c
@@ -866,7 +866,7 @@ static int add_certificate_to_store(TLS_IO_INSTANCE* tls_io_instance, const char
         }
         else
         {
-#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && (OPENSSL_VERSION_NUMBER < 0x20000000L)
+#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && (OPENSSL_VERSION_NUMBER <= 0x30000030L)
             const BIO_METHOD* bio_method;
 #else
             BIO_METHOD* bio_method;

diff --git a/tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/utpm/deps/c-utility/adapters/x509_openssl.c b/tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/utpm/deps/c-utility/adapters/x509_openssl.c
index 11be7a03..96993f40 100644
--- a/tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/utpm/deps/c-utility/adapters/x509_openssl.c
+++ b/tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/utpm/deps/c-utility/adapters/x509_openssl.c
@@ -73,7 +73,7 @@ static int load_certificate_chain(SSL_CTX* ssl_ctx, const char* certificate)
                 // certificates.
 
                 /* Codes_SRS_X509_OPENSSL_07_006: [ If successful x509_openssl_add_ecc_credentials shall to import each certificate in the cert chain. ] */
-#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && (OPENSSL_VERSION_NUMBER < 0x20000000L)
+#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && (OPENSSL_VERSION_NUMBER <= 0x30000030L)
                 SSL_CTX_clear_extra_chain_certs(ssl_ctx);
 #else
                 if (ssl_ctx->extra_certs != NULL)
@@ -264,7 +264,7 @@ int x509_openssl_add_certificates(SSL_CTX* ssl_ctx, const char* certificates)
         else
         {
             /*Codes_SRS_X509_OPENSSL_02_012: [ x509_openssl_add_certificates shall get the memory BIO method function by calling BIO_s_mem. ]*/
-#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && (OPENSSL_VERSION_NUMBER < 0x20000000L)
+#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && (OPENSSL_VERSION_NUMBER <= 0x30000030L)
             const BIO_METHOD* bio_method;
 #else
             BIO_METHOD* bio_method;
-- 
2.30.2
