diff --git a/cert/aziot-certd/src/lib.rs b/cert/aziot-certd/src/lib.rs
index 0956775..0b0cb13 100644
--- a/cert/aziot-certd/src/lib.rs
+++ b/cert/aziot-certd/src/lib.rs
@@ -51,6 +51,8 @@ pub async fn main(
     config_path: PathBuf,
     config_directory_path: PathBuf,
 ) -> Result<(Incoming, http::Service), Box<dyn StdError>> {
+    openssl::provider::Provider::try_load(None, "default", true)?;
+
     let Config {
         homedir_path,
         max_requests,
diff --git a/identity/aziot-identityd/src/lib.rs b/identity/aziot-identityd/src/lib.rs
index 78d95f0..b6b91b5 100644
--- a/identity/aziot-identityd/src/lib.rs
+++ b/identity/aziot-identityd/src/lib.rs
@@ -61,6 +61,8 @@ pub async fn main(
     config_path: std::path::PathBuf,
     config_directory_path: std::path::PathBuf,
 ) -> Result<(http_common::Incoming, http::Service), Box<dyn std::error::Error>> {
+    openssl::provider::Provider::try_load(None, "default", true)?;
+
     let settings = settings.check().map_err(InternalError::BadSettings)?;
 
     let homedir_path = &settings.homedir;
