diff --git a/git/tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/utpm/deps/umock-c/inc/umock_c/umock_c.h b/git/tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/utpm/deps/umock-c/inc/umock_c/umock_c.h
index 7b65735..85c2cb1 100644
--- a/git/tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/utpm/deps/umock-c/inc/umock_c/umock_c.h
+++ b/git/tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/utpm/deps/umock-c/inc/umock_c/umock_c.h
@@ -7,7 +7,7 @@
 #ifdef __cplusplus
 #include <cstdlib>
 #else
-#include <stdlib.h>
+//#include <stdlib.h>
 #endif

 #include "azure_macro_utils/macro_utils.h"
diff --git a/git/tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/utpm/deps/umock-c/inc/umock_c/umock_c_internal.h b/git/tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/utpm/deps/umock-c/inc/umock_c/umock_c_internal.h
index d3ee8c2..27a9a62 100644
--- a/git/tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/utpm/deps/umock-c/inc/umock_c/umock_c_internal.h
+++ b/git/tpm/aziot-tpm-sys/azure-iot-hsm-c/deps/utpm/deps/umock-c/inc/umock_c/umock_c_internal.h
@@ -8,7 +8,7 @@
 #include <cstdlib>
 #include <cstdio>
 #else
-#include <stdlib.h>
+//#include <stdlib.h>
 #include <stdio.h>
 #endif
