From 8c534d0c2b1d500a1dd55915e93d5480c5fe6fad Mon Sep 17 00:00:00 2001
From: Kas User <kas@example.com>
Date: Fri, 28 Jul 2023 16:09:41 +0000
Subject: [PATCH 1/1] usb:xhci-ring.c: don't queue URB for halted endpoint

Signed-off-by: Kas User <kas@example.com>
---
 drivers/usb/host/xhci-ring.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/host/xhci-ring.c b/drivers/usb/host/xhci-ring.c
index acf437104a..6d995f446e 100644
--- a/drivers/usb/host/xhci-ring.c
+++ b/drivers/usb/host/xhci-ring.c
@@ -227,7 +227,8 @@ static int prepare_ring(struct xhci_ctrl *ctrl, struct xhci_ring *ep_ring,
 		puts("WARN waiting for error on ep to be cleared\n");
 		return -EINVAL;
 	case EP_STATE_HALTED:
-		puts("WARN halted endpoint, queueing URB anyway.\n");
+		puts("WARN halted endpoint.\n");
+		return -EPIPE;
 	case EP_STATE_STOPPED:
 	case EP_STATE_RUNNING:
 		debug("EP STATE RUNNING.\n");
-- 
2.30.2

