From a23ea936267e559ab6c69ea75cbf023eb2dcfcfd Mon Sep 17 00:00:00 2001
From: Kas User <kas@example.com>
Date: Wed, 30 Mar 2022 14:13:57 +0000
Subject: [PATCH] disable autoload replaced by more flexible pxe boot

---
 include/configs/rpi.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/configs/rpi.h b/include/configs/rpi.h
index 9bbb7f40d7..f58120dd16 100644
--- a/include/configs/rpi.h
+++ b/include/configs/rpi.h
@@ -66,6 +66,9 @@
 /* Console configuration */
 #define CONFIG_SYS_CBSIZE		1024
 
+/* use pxe boot instead */
+#define CONFIG_SYS_AUTOLOAD		"no"
+
 /* Environment */
 
 /* Shell */
