diff --git a/include/configs/rpi.h b/include/configs/rpi.h
index 4c5c1ac31f..a4b648fb9e 100644
--- a/include/configs/rpi.h
+++ b/include/configs/rpi.h
@@ -13,6 +13,8 @@
 #include <asm/arch/base.h>
 #endif

+#include <configs/omnect_env.h>
+
 /* Architecture, CPU, etc.*/

 /* Use SoC timer for AArch32, but architected timer for AArch64 */
@@ -84,6 +86,10 @@
 #define INITRD_HIGH "ffffffff"
 #endif

+#ifdef CONFIG_ENV_IS_IN_MMC
+#define CONFIG_SYS_MMC_ENV_DEV 0
+#endif
+
 /*
  * Memory layout for where various images get loaded by boot scripts:
  *
@@ -174,7 +180,11 @@
 	ENV_DEVICE_SETTINGS \
 	ENV_DFU_SETTINGS \
 	ENV_MEM_LAYOUT_SETTINGS \
-	BOOTENV
+	BOOTENV \
+	OMNECT_ENV_UPDATE_WORKFLOW \
+	OMNECT_ENV_SETTINGS \
+	OMNECT_ENV_BOOTARGS


 #endif
