diff --git a/include/configs/rpi.h b/include/configs/rpi.h
index 4c5c1ac3..b24226cc 100644
--- a/include/configs/rpi.h
+++ b/include/configs/rpi.h
@@ -13,6 +13,8 @@
 #include <asm/arch/base.h>
 #endif
 
+#include <configs/omnect_env.h>
+
 /* Architecture, CPU, etc.*/
 
 /* Use SoC timer for AArch32, but architected timer for AArch64 */
@@ -174,7 +176,10 @@
 	ENV_DEVICE_SETTINGS \
 	ENV_DFU_SETTINGS \
 	ENV_MEM_LAYOUT_SETTINGS \
-	BOOTENV
+	BOOTENV \
+	OMNECT_ENV_UPDATE_WORKFLOW \
+	OMNECT_ENV_SETTINGS \
+	OMNECT_ENV_BOOTARGS
 
 
 #endif
