diff --git a/board/raspberrypi/rpi/rpi.env b/board/raspberrypi/rpi/rpi.env
--- a/board/raspberrypi/rpi/rpi.env
+++ b/board/raspberrypi/rpi/rpi.env
@@ -2,8 +2,6 @@
 
 /* environment for Raspberry Pi boards */
 
-dhcpuboot=usb start; dhcp u-boot.uimg; bootm
-
 /* Environment */
 stdin=serial,usbkbd
 stdout=serial,vidconsole
@@ -77,4 +75,4 @@
 fdt_addr_r=0x05600000
 ramdisk_addr_r=0x05700000
 
-boot_targets=mmc usb pxe dhcp
+#include <env/omnect_env.env>
