diff --git a/board/raspberrypi/rpi/rpi.env b/board/raspberrypi/rpi/rpi.env
index 30228285ed..b9765c4219 100644
--- a/board/raspberrypi/rpi/rpi.env
+++ b/board/raspberrypi/rpi/rpi.env
@@ -2,8 +2,6 @@
 
 /* environment for Raspberry Pi boards */
 
-dhcpuboot=usb start; dhcp u-boot.uimg; bootm
-
 /* Environment */
 stdin=serial,usbkbd
 stdout=serial,vidconsole
@@ -74,4 +72,4 @@ pxefile_addr_r=0x02500000
 fdt_addr_r=0x02600000
 ramdisk_addr_r=0x02700000
 
-boot_targets=mmc usb pxe dhcp
+#include <env/omnect_env.env>
