From 16e05f9078eb56b1cea3c2ac0d90029ece0337fa Mon Sep 17 00:00:00 2001
From: SteffenRumler <steffen.rumler@conplement.de>
Date: Tue, 30 Nov 2021 13:43:10 +0100
Subject: [PATCH] Fix inappropriate fifo truncate

---
 scp.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/scp.c b/scp.c
index 742ae00..be1b785 100644
--- a/scp.c
+++ b/scp.c
@@ -1051,7 +1051,8 @@ bad:			run_err("%s: %s", np, strerror(errno));
 			wrerr = YES;
 			wrerrno = errno;
 		}
-		if (wrerr == NO && ftruncate(ofd, size) != 0) {
+		if (wrerr == NO && (!exists || S_ISREG(stb.st_mode)) &&
+		    ftruncate(ofd, size) != 0) {
 			run_err("%s: truncate: %s", np, strerror(errno));
 			wrerr = DISPLAYED;
 		}
