diff --git a/src/utils/eis_utils/src/eis_utils.c b/src/utils/eis_utils/src/eis_utils.c
index 85cc8f2..fc4dcf2 100644
--- a/src/utils/eis_utils/src/eis_utils.c
+++ b/src/utils/eis_utils/src/eis_utils.c
@@ -633,6 +633,11 @@ EISUtilityResult RequestConnectionStringFromEISWithExpiry(
     result.err = EISErr_Ok;
 done:
 
+#ifndef NDEBUG
+    printf("identity json: %s\r\n", identityResponseStr);
+    printf("connection string: %s\r\n", connectionStr);
+#endif
+
     json_value_free(identityResponseJson);
 
     free(resourceUri);
