diff --git a/cmake/agentRules.cmake b/cmake/agentRules.cmake
index 68e9c41..c693fcd 100644
--- a/cmake/agentRules.cmake
+++ b/cmake/agentRules.cmake
@@ -40,6 +40,19 @@ function (target_link_iothub_client target scope)
         iothub_client_http_transport
         uhttp)
 
+    #depending on your azure-iot-sdk-c config options you have to link more:
+    find_package (OpenSSL REQUIRED)
+    target_link_libraries (
+        ${target}
+        ${scope}
+        prov_auth_client
+        hsm_security_client
+        msr_riot
+        utpm
+        dl
+        OpenSSL::SSL)
+
+
 endfunction (target_link_iothub_client)
 
 function (target_link_digital_twin_client target scope)
