From c63cd3f2f4f9125a6ef2043617a7229a32bed780 Mon Sep 17 00:00:00 2001
From: Kas User <kas@example.com>
Date: Thu, 9 Jun 2022 14:59:57 +0000
Subject: [PATCH] workaround deprecated declarations openssl3

---
 src/utils/crypto_utils/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/utils/crypto_utils/CMakeLists.txt b/src/utils/crypto_utils/CMakeLists.txt
index 423e8eb..9538798 100644
--- a/src/utils/crypto_utils/CMakeLists.txt
+++ b/src/utils/crypto_utils/CMakeLists.txt
@@ -15,6 +15,9 @@ find_package (OpenSSL REQUIRED)
 #
 set_property (TARGET ${PROJECT_NAME} PROPERTY POSITION_INDEPENDENT_CODE ON)
 
+# workaround for deprecated declarations enforced by openssl 3.0.X (used by Yocto kirkstone)
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-deprecated-declarations")
+
 target_link_libraries (
     ${PROJECT_NAME}
     PUBLIC aduc::c_utils
