diff --git a/arch/arm/boot/dts/overlays/ramoops-pi4-overlay.dts b/arch/arm/boot/dts/overlays/ramoops-pi4-overlay.dts
index 4f3d30ef0..c3c5ab565 100644
--- a/arch/arm/boot/dts/overlays/ramoops-pi4-overlay.dts
+++ b/arch/arm/boot/dts/overlays/ramoops-pi4-overlay.dts
@@ -9,9 +9,12 @@ fragment@0 {
 		__overlay__ {
 			ramoops: ramoops@b000000 {
 				compatible = "ramoops";
-				reg = <0x0 0x0b000000 0x10000>; /* 64kB */
+				reg = <0x0 0x0b000000 0x40000>; /* 256kB */
 				record-size = <0x4000>; /* 16kB */
 				console-size = <0>; /* disabled by default */
+				ecc-size = <0>; /* disabled by default */
+				pmsg-size = <0>; /* disabled by default */
+				ftrace-size = <0>; /* disabled by default */
 			};
 		};
 	};
@@ -21,5 +24,8 @@ __overrides__ {
 		total-size = <&ramoops>,"reg:8";
 		record-size = <&ramoops>,"record-size:0";
 		console-size = <&ramoops>,"console-size:0";
+		ecc-size = <&ramoops>,"ecc-size:0";
+		pmsg-size = <&ramoops>,"pmsg-size:0";
+		ftrace-size = <&ramoops>,"ftrace-size:0";
 	};
 };
