diff --git a/aziotctl/aziotctl-common/Cargo.toml b/aziotctl/aziotctl-common/Cargo.toml
index 7fa3832..b6270cb 100644
--- a/aziotctl/aziotctl-common/Cargo.toml
+++ b/aziotctl/aziotctl-common/Cargo.toml
@@ -29,3 +29,6 @@ toml = "0.7"
 
 [features]
 snapctl = []
+
+[lints.rust]
+unexpected_cfgs = { level = "warn", check-cfg = ['cfg(target_pointer_width,values("128"))'] }
diff --git a/identity/aziot-identity-common/src/lib.rs b/identity/aziot-identity-common/src/lib.rs
index 3a49e4e..24942da 100644
--- a/identity/aziot-identity-common/src/lib.rs
+++ b/identity/aziot-identity-common/src/lib.rs
@@ -98,7 +98,7 @@ impl std::fmt::Display for AuthenticationType {
     }
 }
 
-pub struct Uid(u32);
+pub struct Uid();
 
 #[derive(Clone, Debug, Eq, PartialEq, PartialOrd, serde::Deserialize, serde::Serialize)]
 #[serde(rename_all = "lowercase")]
diff --git a/openssl-sys2/Cargo.toml b/openssl-sys2/Cargo.toml
index dad9eeb..9f7f1db 100644
--- a/openssl-sys2/Cargo.toml
+++ b/openssl-sys2/Cargo.toml
@@ -11,3 +11,6 @@ openssl-sys = "0.9"
 
 [build-dependencies]
 openssl-build = { path = "../openssl-build/" }
+
+[lints.rust]
+unexpected_cfgs = { level = "warn", check-cfg = ['cfg(ossl110)'] }
diff --git a/openssl-sys2/src/lib.rs b/openssl-sys2/src/lib.rs
index 595c4e7..24d2d87 100644
--- a/openssl-sys2/src/lib.rs
+++ b/openssl-sys2/src/lib.rs
@@ -11,7 +11,6 @@ mod ec;
 pub use ec::*;
 
 mod ecdsa;
-pub use ecdsa::*;
 
 mod engine;
 pub use engine::*;
diff --git a/openssl2/Cargo.toml b/openssl2/Cargo.toml
index 666c181..316e34b 100644
--- a/openssl2/Cargo.toml
+++ b/openssl2/Cargo.toml
@@ -18,3 +18,6 @@ openssl-sys2 = { path = "../openssl-sys2/" }
 
 [build-dependencies]
 openssl-build = { path = "../openssl-build/" }
+
+[lints.rust]
+unexpected_cfgs = { level = "warn", check-cfg = ['cfg(ossl110)'] }
