diff --git a/edgelet/docker-rs/Cargo.toml b/edgelet/docker-rs/Cargo.toml
index f2ef566db..97eb26311 100644
--- a/edgelet/docker-rs/Cargo.toml
+++ b/edgelet/docker-rs/Cargo.toml
@@ -18,3 +18,6 @@ url = "2"
 tokio = { version = "1", features = ["macros", "rt"] }
 
 edgelet-test-utils = { path = "../edgelet-test-utils" }
+
+[lints.rust]
+unexpected_cfgs = { level = "warn", check-cfg = ['cfg(tarpaulin_include)'] }
\ No newline at end of file
diff --git a/edgelet/iotedge/Cargo.toml b/edgelet/iotedge/Cargo.toml
index e70c50722..e2274db68 100644
--- a/edgelet/iotedge/Cargo.toml
+++ b/edgelet/iotedge/Cargo.toml
@@ -58,3 +58,6 @@ support-bundle = { path = "../support-bundle" }
 
 [features]
 snapctl = ["aziotctl-common/snapctl", "support-bundle/snapctl"]
+
+[lints.rust]
+unexpected_cfgs = { level = "warn", check-cfg = ['cfg(contenttrust)'] }
\ No newline at end of file
diff --git a/edgelet/iotedge/src/config/import/old_config/provisioning.rs b/edgelet/iotedge/src/config/import/old_config/provisioning.rs
index 1e033b9ad..03b8b9eb0 100644
--- a/edgelet/iotedge/src/config/import/old_config/provisioning.rs
+++ b/edgelet/iotedge/src/config/import/old_config/provisioning.rs
@@ -18,7 +18,7 @@ pub(crate) struct Provisioning {
 pub(crate) enum ProvisioningType {
     Manual(Manual),
     Dps(Dps),
-    External(External),
+    External(()),
 }
 
 #[derive(Debug)]
