From 93736d1a831264c6b104b6dc3a309b1b6f8d9211 Mon Sep 17 00:00:00 2001
From: Joerg Zeidler <joerg.zeidler@conplement.de>
Date: Tue, 9 Feb 2021 12:49:54 +0000
Subject: [PATCH] generate patch

---
 edgelet/contrib/systemd/debian/iotedge.service | 1 +
 1 file changed, 1 insertion(+)

diff --git a/edgelet/contrib/systemd/debian/iotedge.service b/edgelet/contrib/systemd/debian/iotedge.service
index 6cbbbe7..eb0821e 100644
--- a/edgelet/contrib/systemd/debian/iotedge.service
+++ b/edgelet/contrib/systemd/debian/iotedge.service
@@ -4,6 +4,7 @@ After=network-online.target docker.service iotedge.socket iotedge.mgmt.socket
 Requires=iotedge.socket iotedge.mgmt.socket
 Wants=network-online.target docker.service
 Documentation=man:iotedged(8)
+ConditionPathExists=/etc/ics_dm/enrolled
 
 [Service]
 ExecStart=/usr/bin/iotedged -c /etc/iotedge/config.yaml
-- 
2.20.1

