diff --git a/include/configs/phycore_imx8mm.h b/include/configs/phycore_imx8mm.h
index 57f467b1..9a53fec1 100644
--- a/include/configs/phycore_imx8mm.h
+++ b/include/configs/phycore_imx8mm.h
@@ -11,6 +11,8 @@
 #include <linux/stringify.h>
 #include <asm/arch/imx-regs.h>
 
+#include <configs/omnect_env.h>
+
 #include "phycore_rauc_env.h"
 #include "phycore_fitimage_env.h"
 
@@ -54,6 +56,7 @@
 	"ramdisk_addr_r=0x49200000\0" \
 	"scan_boot_script=if test -e mmc ${devnum}:1 /boot.scr; then echo Found U-Boot script; run boot_script; echo SCRIPT FAILED: continuing...; fi\0" \
 	"script_addr_r=0x49100000\0" \
+	OMNECT_ENV_SETTINGS \
 
 #ifdef CONFIG_IMX_HAB
 #define BOOTCOMMAND_APPEND "reset;"
