diff --git a/board/phytec/phycore_imx8mm/spl.c b/board/phytec/phycore_imx8mm/spl.c
index 0dfc0da4..4d79c0a5 100644
--- a/board/phytec/phycore_imx8mm/spl.c
+++ b/board/phytec/phycore_imx8mm/spl.c
@@ -331,6 +331,8 @@ int board_early_init_f(void)
 
 	init_uart_clk(2);
 
+	gd->flags |= (GD_FLG_SILENT | GD_FLG_DISABLE_CONSOLE);
+
 	return 0;
 }
 
diff --git a/include/configs/phycore_imx8mm.h b/include/configs/phycore_imx8mm.h
index 57f467b1..6338d974 100644
--- a/include/configs/phycore_imx8mm.h
+++ b/include/configs/phycore_imx8mm.h
@@ -43,6 +43,7 @@
 	"load_fdt_script=fatload mmc ${devnum}:1 ${script_addr_r} /fdt-load.scr; source ${script_addr_r}\0" \
 	"bootcmd_pxe=run load_fdt_script; setenv kernel_addr_r ${loadaddr}; dhcp; if pxe get; then pxe boot; fi\0" \
 	"bootargs=console=ttymxc2,115200\0" \
+	"bootdelay=-2\0" \
 	"bootpart=2\0" \
 	"devnum=" __stringify(CONFIG_SYS_MMC_ENV_DEV) "\0" \
 	"devtype=mmc\0" \
@@ -54,6 +55,7 @@
 	"ramdisk_addr_r=0x49200000\0" \
 	"scan_boot_script=if test -e mmc ${devnum}:1 /boot.scr; then echo Found U-Boot script; run boot_script; echo SCRIPT FAILED: continuing...; fi\0" \
 	"script_addr_r=0x49100000\0" \
+	"silent=1\0" \
 
 #ifdef CONFIG_IMX_HAB
 #define BOOTCOMMAND_APPEND "reset;"
