diff --git a/common/bootm.c b/common/bootm.c
index 902c13880d..0825b1eeb5 100644
--- a/common/bootm.c
+++ b/common/bootm.c
@@ -28,8 +28,8 @@
 #include <image.h>
 
 #ifndef CONFIG_SYS_BOOTM_LEN
-/* use 8MByte as default max gunzip size */
-#define CONFIG_SYS_BOOTM_LEN	0x800000
+/* use 64MByte as default max gunzip size */
+#define CONFIG_SYS_BOOTM_LEN	0x4000000
 #endif
 
 #define IH_INITRD_ARCH IH_ARCH_DEFAULT
