From 40fe0095546fad09645173c51847925cd8ee80a5 Mon Sep 17 00:00:00 2001
From: Kas User <kas@example.com>
Date: Wed, 30 Mar 2022 13:49:34 +0000
Subject: [PATCH] use 64MByte as default max gunzip size

---
 boot/bootm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/boot/bootm.c b/boot/bootm.c
index 4482f84b40..dde31f60d1 100644
--- a/boot/bootm.c
+++ b/boot/bootm.c
@@ -34,8 +34,8 @@
 #include <image.h>
 
 #ifndef CONFIG_SYS_BOOTM_LEN
-/* use 8MByte as default max gunzip size */
-#define CONFIG_SYS_BOOTM_LEN	0x800000
+/* use 64MByte as default max gunzip size */
+#define CONFIG_SYS_BOOTM_LEN	0x4000000
 #endif
 
 #define MAX_CMDLINE_SIZE	SZ_4K
