diff --git a/include/env_default.h b/include/env_default.h
--- a/include/env_default.h
+++ b/include/env_default.h
@@ -12,6 +12,7 @@
 #include <linux/stringify.h>
 
 #include <generated/environment.h>
+#include <configs/omnect_env.h>
 
 #ifdef DEFAULT_ENV_INSTANCE_EMBEDDED
 env_t embedded_environment __UBOOT_ENV_SECTION__(environment) = {
diff --git a/include/env_flags.h b/include/env_flags.h
--- a/include/env_flags.h
+++ b/include/env_flags.h
@@ -9,6 +9,8 @@
 
+#include <configs/omnect_env.h>
+
 enum env_flags_vartype {
 	env_flags_vartype_string,
 	env_flags_vartype_decimal,
 	env_flags_vartype_hex,
 	env_flags_vartype_bool,
