diff -Naur 0.2.3-r0.orig/azure-iot-sdk/Cargo.toml 0.2.3-r0/azure-iot-sdk/Cargo.toml
--- 0.2.3-r0.orig/azure-iot-sdk/Cargo.toml	2022-02-27 14:51:51.670927150 +0100
+++ 0.2.3-r0/azure-iot-sdk/Cargo.toml	2022-02-27 14:46:26.986932954 +0100
# Manually adapted: fetch eis-utils with git submodule fetcher
@@ -1,15 +1,15 @@
 [package]
 name = "azure-iot-sdk"
 version = "0.5.0"
-edition = "2021"
+edition = "2018"
 authors = ["Joerg Zeidler <joerg.zeidler@conplement.de>", "Jan Zachmann <jan.zachmann@conplement.de>"]
 repository = "git@github.com:ICS-DeviceManagement/azure-iot-sdk.git"
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-azure-iot-sdk-sys = { git = "ssh://git@github.com/ICS-DeviceManagement/azure-iot-sdk-sys.git", tag = "0.2.1" }
-eis-utils = { git = "ssh://git@github.com/ICS-DeviceManagement/eis-utils.git", tag = "0.2.0" }
+azure-iot-sdk-sys = { path = "../azure-iot-sdk-sys" }
+eis-utils = { path = "../eis-utils" }
 serde_json = "1.0"
 log = "0.4"
 rand = "0.8"
diff -Naur 0.2.3-r0.orig/azure-iot-sdk-sys/Cargo.toml 0.2.3-r0/azure-iot-sdk-sys/Cargo.toml
--- 0.2.3-r0.orig/azure-iot-sdk-sys/Cargo.toml	2022-02-27 14:51:51.670927150 +0100
+++ 0.2.3-r0/azure-iot-sdk-sys/Cargo.toml	2022-02-27 14:48:11.966931077 +0100
@@ -1,7 +1,7 @@
 [package]
 name = "azure-iot-sdk-sys"
 version = "0.2.1"
-edition = "2021"
+edition = "2018"
 authors = ["Joerg Zeidler <joerg.zeidler@conplement.de>", "Jan Zachmann <jan.zachmann@conplement.de>"]
 build = "build.rs"
 repository = "https://github.com/ICS-DeviceManagement/azure-iot-sdk-sys"
diff -Naur 0.2.3-r0.orig/git/Cargo.toml 0.2.3-r0/git/Cargo.toml
--- 0.2.3-r0.orig/git/Cargo.toml	2022-02-27 14:36:13.938943000 +0100
+++ 0.2.3-r0/git/Cargo.toml	2022-02-27 14:48:09.258931126 +0100
@@ -1,7 +1,7 @@
 [package]
 name = "iot-client-template-rs"
 version = "0.2.3"
-edition = "2021"
+edition = "2018"
 authors = ["Joerg Zeidler <joerg.zeidler@conplement.de>", "Jan Zachmann <jan.zachmann@conplement.de>"]
 repository = "git@github.com:ICS-DeviceManagement/iot-client-template-rs.git"
 license = "MIT OR Apache-2.0"
@@ -9,11 +9,11 @@
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-azure-iot-sdk = { git = "ssh://git@github.com/ICS-DeviceManagement/azure-iot-sdk.git", tag = "0.5.0" }
+azure-iot-sdk = { path = "../azure-iot-sdk" }
 serde_json = "1.0"
 env_logger = "0.8"
 log = "0.4"
-sd-notify = { git = "ssh://git@github.com/ICS-DeviceManagement/sd-notify.git", tag = "v0.5.0", optional = true }
+sd-notify = { path = "../sd-notify" , optional = true }
 
 
 [features]
diff -Naur 0.2.3-r0.orig/sd-notify/Cargo.toml 0.2.3-r0/sd-notify/Cargo.toml
--- 0.2.3-r0.orig/sd-notify/Cargo.toml	2022-02-27 14:36:13.938943000 +0100
+++ 0.2.3-r0/sd-notify/Cargo.toml	2022-02-27 14:48:34.950930666 +0100
@@ -1,7 +1,7 @@
 [package]
 name = "sd-notify"
 version = "0.5.0"
-edition = "2021"
+edition = "2018"
 description = "Lightweight crate for systemd service state notifications"
 readme = "README.md"
 keywords = ["systemd", "sd_notify"]
