diff --git a/edgelet/edgelet-http-mgmt/src/error.rs b/edgelet/edgelet-http-mgmt/src/error.rs
index 71e2f544b..2b8aa0d62 100644
--- a/edgelet/edgelet-http-mgmt/src/error.rs
+++ b/edgelet/edgelet-http-mgmt/src/error.rs
@@ -125,7 +125,7 @@ impl From<Context<ErrorKind>> for Error {
 impl IntoResponse for Error {
     fn into_response(self) -> Response<Body> {
         let mut message = self.to_string();
-        for cause in Fail::iter_causes(&self) {
+        for cause in <dyn Fail>::iter_causes(&self) {
             message.push_str(&format!("\n\tcaused by: {}", cause));
         }
 
