diff --git a/arch/arm64/boot/dts/freescale/imx8mm-phygate-tauri.dts b/arch/arm64/boot/dts/freescale/imx8mm-phygate-tauri.dts
index 773e0a4db..9402efe75 100644
--- a/arch/arm64/boot/dts/freescale/imx8mm-phygate-tauri-l.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mm-phygate-tauri-l.dts
@@ -28,6 +28,18 @@ can_osc_40m: can-clock@0 {
 		};
 	};
 
+	reserved-memory {
+		ramoops: ramoops@b0000000 {
+			compatible = "ramoops";
+			reg = <0x0 0x0b0000000 0x0 0x40000>; /*	   256kB */
+			record-size = <0x1000>;              /*	     4kB */
+			console-size = <0x10000>;            /*	     64k */
+			ecc-size = <1>;                      /*  enabled */
+			pmsg-size = <0x4000>;                /*     16kB */
+			ftrace-size = <0>;                   /* disabled */
+		};
+	};
+
 	gpio-keys {
 		compatible = "gpio-keys";
 		pinctrl-names = "default";
