diff --git a/Cargo.lock b/Cargo.lock
index 232d951..ed52970 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -266,6 +266,7 @@ dependencies = [
  "openssl2",
  "percent-encoding",
  "regex",
+ "sd-notify",
  "serde",
  "serde_json",
  "tokio",
@@ -2200,6 +2201,12 @@ version = "1.1.0"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "d29ab0c6d3fc0ee92fe66e2d99f700eab17a8d57d1c1d3b748380fb20baa78cd"
 
+[[package]]
+name = "sd-notify"
+version = "0.4.1"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "621e3680f3e07db4c9c2c3fb07c6223ab2fab2e54bd3c04c3ae037990f428c32"
+
 [[package]]
 name = "serde"
 version = "1.0.164"
diff --git a/identity/aziot-identityd/Cargo.toml b/identity/aziot-identityd/Cargo.toml
index fd0eb3d..e940e38 100644
--- a/identity/aziot-identityd/Cargo.toml
+++ b/identity/aziot-identityd/Cargo.toml
@@ -23,6 +23,7 @@ openssl = "0.10"
 openssl-sys = "0.9"
 percent-encoding = "2"
 regex = "1"
+sd-notify = "0.4"
 serde = "1"
 serde_json = "1.0"
 tokio = { version = "1", features = ["parking_lot", "time"] }
diff --git a/identity/aziot-identityd/aziot-identityd.service.in b/identity/aziot-identityd/aziot-identityd.service.in
index 42adf6f..e9f3b9d 100644
--- a/identity/aziot-identityd/aziot-identityd.service.in
+++ b/identity/aziot-identityd/aziot-identityd.service.in
@@ -4,6 +4,7 @@ Requires=aziot-identityd.socket
 After=aziot-identityd.socket
 
 [Service]
+Type=notify
 Environment="LD_LIBRARY_PATH=@private-libs@"
 ExecStart=@libexecdir@/aziot-identity-service/aziot-identityd
 KillMode=process
diff --git a/identity/aziot-identityd/src/lib.rs b/identity/aziot-identityd/src/lib.rs
index f6ed80f..62cb9ff 100644
--- a/identity/aziot-identityd/src/lib.rs
+++ b/identity/aziot-identityd/src/lib.rs
@@ -158,6 +158,8 @@ pub async fn main(
 
     let service = http::Service { api };
 
+    let _ = sd_notify::notify(false, &[sd_notify::NotifyState::Ready]);
+
     let incoming = connector
         .incoming(http_common::SOCKET_DEFAULT_PERMISSION, max_requests, None)
         .await?;
