diff --git a/.cargo/config.toml b/.cargo/config.toml
deleted file mode 100644
index 5572620..0000000
--- a/.cargo/config.toml
+++ /dev/null
@@ -1,15 +0,0 @@
-[env]
-USER_AZIOTID = "aziotid"
-USER_AZIOTKS = "aziotks"
-USER_AZIOTCS = "aziotcs"
-USER_AZIOTTPM = "aziottpm"
-SOCKET_DIR = "/run/aziot"
-
-[registries]
-iotedge_PublicPackages = { index = "sparse+https://pkgs.dev.azure.com/iotedge/iotedge/_packaging/iotedge_PublicPackages/Cargo/index/" }
-
-[registry]
-global-credential-providers = ["cargo:token", "cargo:libsecret"]
-
-[source.crates-io]
-replace-with = "iotedge_PublicPackages"
diff --git a/Cargo.lock b/Cargo.lock
index bcde79e..fe5d4c3 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -302,6 +302,7 @@ dependencies = [
  "openssl2",
  "percent-encoding",
  "regex",
+ "sd-notify",
  "serde",
  "serde_json",
  "tokio",
@@ -2224,6 +2225,12 @@ dependencies = [
  "serde",
 ]
 
+[[package]]
+name = "sd-notify"
+version = "0.4.1"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "621e3680f3e07db4c9c2c3fb07c6223ab2fab2e54bd3c04c3ae037990f428c32"
+
 [[package]]
 name = "pkcs11-sys"
 version = "0.1.0"
diff --git a/identity/aziot-identityd/Cargo.toml b/identity/aziot-identityd/Cargo.toml
index 324cd7e..eac5cf5 100644
--- a/identity/aziot-identityd/Cargo.toml
+++ b/identity/aziot-identityd/Cargo.toml
@@ -21,6 +21,7 @@ openssl = { workspace = true }
 openssl-sys = { workspace = true }
 percent-encoding = { workspace = true }
 regex = { workspace = true }
+sd-notify = "0.4"
 serde = { workspace = true }
 serde_json = { workspace = true }
 tokio = { workspace = true }
diff --git a/identity/aziot-identityd/aziot-identityd.service.in b/identity/aziot-identityd/aziot-identityd.service.in
index 7e261fd..ecf47ac 100644
--- a/identity/aziot-identityd/aziot-identityd.service.in
+++ b/identity/aziot-identityd/aziot-identityd.service.in
@@ -4,6 +4,7 @@ Requires=aziot-identityd.socket
 After=aziot-identityd.socket
 
 [Service]
+Type=notify
 Environment="LD_LIBRARY_PATH=@private-libs@"
 ExecStart=@libexecdir@/aziot-identity-service/aziot-identityd
 KillMode=process
diff --git a/identity/aziot-identityd/src/lib.rs b/identity/aziot-identityd/src/lib.rs
index 9237a8d..3a3183d 100644
--- a/identity/aziot-identityd/src/lib.rs
+++ b/identity/aziot-identityd/src/lib.rs
@@ -141,6 +141,8 @@ pub async fn main(
 
     let service = http::Service { api };
 
+    let _ = sd_notify::notify(false, &[sd_notify::NotifyState::Ready]);
+
     let incoming = connector
         .incoming(http_common::SOCKET_DEFAULT_PERMISSION, max_requests, None)
         .await?;
