diff --git a/client-lite/CMakeLists.txt b/client-lite/CMakeLists.txt
index bee455f..0bb0024 100644
--- a/client-lite/CMakeLists.txt
+++ b/client-lite/CMakeLists.txt
@@ -116,6 +116,7 @@ target_link_libraries(${DOSVC_BIN_NAME}
     doversion
     docs_common
     cpprestsdk::cpprest
+    crypto
     ${Boost_LIBRARIES}
     ${Boost_LOG_LIBRARIES})
 
@@ -166,4 +167,4 @@ endif() # Linux
 
 if(DO_BUILD_TESTS)
     add_subdirectory (test)
-endif()
\ No newline at end of file
+endif()
