diff --git a/meta-imx-bsp/conf/layer.conf b/meta-imx-bsp/conf/layer.conf
index 94bea8fe36..5b3d59930f 100644
--- a/meta-imx-bsp/conf/layer.conf
+++ b/meta-imx-bsp/conf/layer.conf
@@ -9,6 +9,7 @@ BBFILE_COLLECTIONS += "fsl-bsp-release"
 BBFILE_PATTERN_fsl-bsp-release := "^${LAYERDIR}/"
 BBFILE_PRIORITY_fsl-bsp-release = "8"
 LAYERSERIES_COMPAT_fsl-bsp-release = "mickledore nanbield scarthgap"
+LAYERDIR_fsl-bsp-release = "${LAYERDIR}"
 
 # Create a mirror for unreliable packages
 MIRRORS += "http://sourceforge.net/.* http://www.nxp.com/lgfiles/updates/other"
