diff --git a/meta-imx-bsp/conf/layer.conf b/meta-imx-bsp/conf/layer.conf
index f5f2f6e57f..16a4420191 100644
--- a/meta-imx-bsp/conf/layer.conf
+++ b/meta-imx-bsp/conf/layer.conf
@@ -8,7 +8,7 @@ BBFILES += "${LAYERDIR}/recipes-*/*/*.bb \
 BBFILE_COLLECTIONS += "fsl-bsp-release"
 BBFILE_PATTERN_fsl-bsp-release := "^${LAYERDIR}/"
 BBFILE_PRIORITY_fsl-bsp-release = "8"
-LAYERSERIES_COMPAT_fsl-bsp-release = "mickledore nanbield"
+LAYERSERIES_COMPAT_fsl-bsp-release = "mickledore nanbield scarthgap"
 
 # Create a mirror for unreliable packages
 MIRRORS += "http://sourceforge.net/.* http://www.nxp.com/lgfiles/updates/other"
@@ -484,3 +484,5 @@ KERNEL_DEVICETREE:append:imx93-14x14-lpddr4x-evk = " \
     freescale/${KERNEL_DEVICETREE_BASENAME}-i3c.dtb \
     freescale/${KERNEL_DEVICETREE_BASENAME}-sja1105.dtb \
 "
+
+LAYERDIR_fsl-bsp-release = "${LAYERDIR}"
diff --git a/meta-imx-sdk/conf/layer.conf b/meta-imx-sdk/conf/layer.conf
index 0627aeba6f..551f160a77 100644
--- a/meta-imx-sdk/conf/layer.conf
+++ b/meta-imx-sdk/conf/layer.conf
@@ -8,7 +8,7 @@ BBFILES += "${LAYERDIR}/recipes-*/*/*.bb \
 BBFILE_COLLECTIONS += "fsl-sdk-release"
 BBFILE_PATTERN_fsl-sdk-release := "^${LAYERDIR}/"
 BBFILE_PRIORITY_fsl-sdk-release = "8"
-LAYERSERIES_COMPAT_fsl-sdk-release = "mickledore nanbield"
+LAYERSERIES_COMPAT_fsl-sdk-release = "mickledore nanbield scarthgap"
 
 BBFILES_DYNAMIC += " \
     chromium-browser-layer:${LAYERDIR}/dynamic-layers/chromium-browser-layer/*/*/*.bb \
