diff --git a/meta-bsp/conf/layer.conf b/meta-bsp/conf/layer.conf
index a161ce65fd..771893d0d5 100644
--- a/meta-bsp/conf/layer.conf
+++ b/meta-bsp/conf/layer.conf
@@ -425,3 +425,5 @@ IMXBOOT_TARGETS_SD:prepend:imx8dxl-ddr3l-evk = \
 
 # Create a mirror for unreliable packages
 MIRRORS += "http://sourceforge.net/.* http://www.nxp.com/lgfiles/updates/other"
+
+LAYERDIR_fsl-bsp-release = "${LAYERDIR}"
