diff --git a/conf/layer.conf b/conf/layer.conf
index 18d100a9..dd2284b1 100644
--- a/conf/layer.conf
+++ b/conf/layer.conf
@@ -33,4 +33,4 @@ BBFILES_DYNAMIC += " \
         stm-st-stm32mp:${LAYERDIR}/dynamic-layers/stm-st-stm32mp/recipes-*/*/*.bbappend \
 "
 
-BBMASK:mx8m-nxp-bsp = "meta-imx/meta-bsp/recipes-graphics/imx-gpu-apitrace/imx-gpu-apitrace_10.0.0.bb"
+BBMASK:append:mx8m-nxp-bsp = "meta-imx/meta-bsp/recipes-graphics/imx-gpu-apitrace/imx-gpu-apitrace_10.0.0.bb"
