diff --git a/meta-networking/recipes-connectivity/networkmanager/networkmanager_1.46.0.bb b/meta-networking/recipes-connectivity/networkmanager/networkmanager_1.46.0.bb
index 8184fcf1a..da5236069 100644
--- a/meta-networking/recipes-connectivity/networkmanager/networkmanager_1.46.0.bb
+++ b/meta-networking/recipes-connectivity/networkmanager/networkmanager_1.46.0.bb
@@ -76,11 +76,11 @@ EXTRA_OEMESON = "\
 CFLAGS:append:libc-musl = " \
     -DRTLD_DEEPBIND=0 \
 "
-do_configure:prepend() {
-    cp -f ${STAGING_LIBDIR}/girepository-1.0/GObject*typelib ${STAGING_LIBDIR_NATIVE}/girepository-1.0/
-    cp -f ${STAGING_LIBDIR}/girepository-1.0/Gio*typelib ${STAGING_LIBDIR_NATIVE}/girepository-1.0/
-    cp -f ${STAGING_LIBDIR}/girepository-1.0/GModule*typelib ${STAGING_LIBDIR_NATIVE}/girepository-1.0/
-}
+# do_configure:prepend() {
+#     cp -f ${STAGING_LIBDIR}/girepository-1.0/GObject*typelib ${STAGING_LIBDIR_NATIVE}/girepository-1.0/
+#     cp -f ${STAGING_LIBDIR}/girepository-1.0/Gio*typelib ${STAGING_LIBDIR_NATIVE}/girepository-1.0/
+#     cp -f ${STAGING_LIBDIR}/girepository-1.0/GModule*typelib ${STAGING_LIBDIR_NATIVE}/girepository-1.0/
+# }
 
 PACKAGECONFIG ??= "readline nss ifupdown dnsmasq nmcli vala \
     ${@bb.utils.contains('DISTRO_FEATURES', 'systemd', 'systemd', bb.utils.contains('DISTRO_FEATURES', 'x11', 'consolekit', '', d), d)} \
diff --git a/meta-oe/recipes-support/nspr/nspr_4.35.bb b/meta-oe/recipes-support/nspr/nspr_4.35.bb
index f8d329398..5db8e2624 100644
--- a/meta-oe/recipes-support/nspr/nspr_4.35.bb
+++ b/meta-oe/recipes-support/nspr/nspr_4.35.bb
@@ -30,7 +30,7 @@ CVE_PRODUCT = "netscape_portable_runtime"
 
 S = "${WORKDIR}/nspr-${PV}/nspr"
 
-RDEPENDS:${PN}-dev += "perl"
+# RDEPENDS:${PN}-dev += "perl"
 TARGET_CC_ARCH += "${LDFLAGS}"
 
 TESTS = " \
@@ -166,9 +166,9 @@ do_compile:prepend() {
 	oe_runmake CROSS_COMPILE=1 CFLAGS="-DXP_UNIX ${BUILD_CFLAGS}" LDFLAGS="" CC="${BUILD_CC}" -C config export
 }
 
-do_compile:append() {
-	oe_runmake -C pr/tests
-}
+# do_compile:append() {
+# 	oe_runmake -C pr/tests
+# }
 
 do_install:append() {
     install -D ${WORKDIR}/nspr.pc.in ${D}${libdir}/pkgconfig/nspr.pc
@@ -180,11 +180,11 @@ do_install:append() {
     -e 's:OEEXECPREFIX:${exec_prefix}:g' \
     ${D}${libdir}/pkgconfig/nspr.pc
 
-    mkdir -p ${D}${libdir}/nspr/tests
-    install -m 0755 ${S}/pr/tests/runtests.pl ${D}${libdir}/nspr/tests
-    install -m 0755 ${S}/pr/tests/runtests.sh ${D}${libdir}/nspr/tests
-    cd ${B}/pr/tests
-    install -m 0755 ${TESTS} ${D}${libdir}/nspr/tests
+    # mkdir -p ${D}${libdir}/nspr/tests
+    # install -m 0755 ${S}/pr/tests/runtests.pl ${D}${libdir}/nspr/tests
+    # install -m 0755 ${S}/pr/tests/runtests.sh ${D}${libdir}/nspr/tests
+    # cd ${B}/pr/tests
+    # install -m 0755 ${TESTS} ${D}${libdir}/nspr/tests
 
     # delete compile-et.pl and perr.properties from ${bindir} because these are
     # only used to generate prerr.c and prerr.h files from prerr.et at compile
