diff --git a/meta/classes/cargo_common.bbclass b/meta/classes/cargo_common.bbclass
index c1bc142d85..362498610d 100644
--- a/meta/classes/cargo_common.bbclass
+++ b/meta/classes/cargo_common.bbclass
@@ -31,12 +31,12 @@ CARGO_RUST_TARGET_CCLD ?= "${RUST_TARGET_CCLD}"
 cargo_common_do_configure () {
 	mkdir -p ${CARGO_HOME}/bitbake
 
-	cat <<- EOF > ${CARGO_HOME}/config
-	# EXTRA_OECARGO_PATHS
-	paths = [
-	$(for p in ${EXTRA_OECARGO_PATHS}; do echo \"$p\",; done)
-	]
-	EOF
+	# cat <<- EOF > ${CARGO_HOME}/config
+	# # EXTRA_OECARGO_PATHS
+	# paths = [
+	# $(for p in ${EXTRA_OECARGO_PATHS}; do echo \"$p\",; done)
+	# ]
+	# EOF
 
 	cat <<- EOF >> ${CARGO_HOME}/config
 
