diff --git a/meta/classes/image.bbclass b/meta/classes/image.bbclass
index 00413d56d1..cef10c5fde 100644
--- a/meta/classes/image.bbclass
+++ b/meta/classes/image.bbclass
@@ -529,27 +529,27 @@ def get_rootfs_size(d):
     size_kb = oe.utils.directory_size(d.getVar("IMAGE_ROOTFS")) / 1024

     base_size = size_kb * overhead_factor
-    bb.debug(1, '%f = %d * %f' % (base_size, size_kb, overhead_factor))
+    bb.verbnote('%f = %d * %f' % (base_size, size_kb, overhead_factor))
     base_size2 = max(base_size, rootfs_req_size) + rootfs_extra_space
-    bb.debug(1, '%f = max(%f, %d)[%f] + %d' % (base_size2, base_size, rootfs_req_size, max(base_size, rootfs_req_size), rootfs_extra_space))
+    bb.verbnote('%f = max(%f, %d)[%f] + %d' % (base_size2, base_size, rootfs_req_size, max(base_size, rootfs_req_size), rootfs_extra_space))

     base_size = base_size2
     if base_size != int(base_size):
         base_size = int(base_size + 1)
     else:
         base_size = int(base_size)
-    bb.debug(1, '%f = int(%f)' % (base_size, base_size2))
+    bb.verbnote('%f = int(%f)' % (base_size, base_size2))

     base_size_saved = base_size
     base_size += rootfs_alignment - 1
     base_size -= base_size % rootfs_alignment
-    bb.debug(1, '%d = aligned(%d)' % (base_size, base_size_saved))
+    bb.verbnote('%d = aligned(%d) [max: %d]' % (base_size, base_size_saved, int(rootfs_maxsize)))

     # Do not check image size of the debugfs image. This is not supposed
     # to be deployed, etc. so it doesn't make sense to limit the size
     # of the debug.
     if (d.getVar('IMAGE_BUILDING_DEBUGFS') or "") == "true":
-        bb.debug(1, 'returning debugfs size %d' % (base_size))
+        bb.verbnote('returning debugfs size %d' % (base_size))
         return base_size

     # Check the rootfs size against IMAGE_ROOTFS_MAXSIZE (if set)
