diff --git a/meta/conf/layer.conf b/meta/conf/layer.conf
index 1f329c3efe..f906531327 100644
--- a/meta/conf/layer.conf
+++ b/meta/conf/layer.conf
@@ -123,3 +123,5 @@ SSTATE_EXCLUDEDEPS_SYSROOT += ".*->autoconf-archive-native"
 # Avoid empty path entries
 BITBAKEPATH := "${@os.path.dirname(bb.utils.which(d.getVar('PATH'),'bitbake'))}"
 PATH := "${@'${BITBAKEPATH}:' if '${BITBAKEPATH}' != '' else ''}${HOSTTOOLS_DIR}"
+
+LAYERDIR_core = "${LAYERDIR}"
