diff --git a/CMakeLists.txt b/CMakeLists.txt
index deeaae7d..0c3321cb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -176,7 +176,7 @@ elseif(LINUX) #LINUX OR APPLE
     endif()
     # Turn off warning that can not be fixed right now
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-variable -Wno-missing-braces -Wno-missing-field-initializers -Wno-format-nonliteral")
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-variable -Wno-missing-braces -Wno-missing-field-initializers -Wno-format-nonliteral")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-variable -Wno-missing-braces -Wno-missing-field-initializers -Wno-format-nonliteral -Wno-deprecated-declarations -Wno-discarded-qualifiers")
 elseif(APPLE)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wformat-security")
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Wformat-security")
diff --git a/adapters/socketio_berkeley.c b/adapters/socketio_berkeley.c
old mode 100755
new mode 100644
diff --git a/adapters/tlsio_openssl.c b/adapters/tlsio_openssl.c
index 4a3df849..2ff848ac 100644
--- a/adapters/tlsio_openssl.c
+++ b/adapters/tlsio_openssl.c
@@ -953,7 +953,7 @@ static int add_certificate_to_store(TLS_IO_INSTANCE* tls_io_instance, const char
         }
         else
         {
-#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && (OPENSSL_VERSION_NUMBER < 0x20000000L)
+#if (OPENSSL_VERSION_NUMBER >= 0x10100000L)
             const BIO_METHOD* bio_method;
 #else
             BIO_METHOD* bio_method;
diff --git a/adapters/x509_openssl.c b/adapters/x509_openssl.c
index 5a9e5ac2..bdb8c6fa 100644
--- a/adapters/x509_openssl.c
+++ b/adapters/x509_openssl.c
@@ -75,7 +75,7 @@ static int load_certificate_chain(SSL_CTX* ssl_ctx, const char* certificate)
                 // certificates.
 
                 /* Codes_SRS_X509_OPENSSL_07_006: [ If successful x509_openssl_add_ecc_credentials shall to import each certificate in the cert chain. ] */
-#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && (OPENSSL_VERSION_NUMBER < 0x20000000L)
+#if (OPENSSL_VERSION_NUMBER >= 0x10100000L)
                 SSL_CTX_clear_extra_chain_certs(ssl_ctx);
 #else
                 if (ssl_ctx->extra_certs != NULL)
@@ -345,7 +345,7 @@ int x509_openssl_add_certificates(SSL_CTX* ssl_ctx, const char* certificates)
         else
         {
             /*Codes_SRS_X509_OPENSSL_02_012: [ x509_openssl_add_certificates shall get the memory BIO method function by calling BIO_s_mem. ]*/
-#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && (OPENSSL_VERSION_NUMBER < 0x20000000L)
+#if (OPENSSL_VERSION_NUMBER >= 0x10100000L)
             const BIO_METHOD* bio_method;
 #else
             BIO_METHOD* bio_method;
diff --git a/src/consolelogger.c b/src/consolelogger.c
old mode 100755
new mode 100644
diff --git a/src/strings.c b/src/strings.c
old mode 100755
new mode 100644
